src/containers/ValetReservation.jsx

```diff
@@ -2,21 +2,40 @@
 import CurrentSpots from '../components/valet-reservation/CurrentSpots.jsx';
 import ReservationList from '../components/valet-reservation/ReservationList.jsx';
 import { Link } from "react-router-dom";
-import { useState } from 'react';
-import postGenerateQrCode from '../utils/postGenerateQrCode.js'
-import getQrCode from '../utils/getQrCode.js';
+import { useState, useEffect, useCallback } from 'react';
+import { postGenerateQrCode, getQrCode, getSpotMetrics, getReservationList } from '../utils/valetReservationUtils.js'
 import { useSelector, useDispatch } from "react-redux";
-import { setDate } from "../utils/slice/valetSlice";
+import { setDate, setSpots, setReservations } from "../utils/slice/valetSlice";
 
 const ValetReservation = () => {
-  //use garage id (1 for testing purposes) and current date (format:DD-MM-YYYY) to get the current reservations
-  //
+  //use garage id (1 for testing purposes) and current date (format:DD-MM-YY) to get the current reservations and spots
   const [showQrCode, setShowQrCode] = useState(false);
   const [qrCodeLink, setQrCodeLink] = useState('');
 
-  const { date } = useSelector((state) => state.valet);
+  const { date, time, garage_id } = useSelector((state) => state.valet);
   const dispatch = useDispatch();
 
+  const fetchSpotsAndReservations = useCallback(async () => {
+    //combining the two fetches into one function; might need to separate them upon further testing
+    try {
+      const spotData = await getSpotMetrics(garage_id, date, time);
+      const reservationData = await getReservationList(garage_id, date);
+      dispatch(setSpots(spotData));
+      dispatch(setReservations(reservationData))
+    } catch (error) {
+      console.log(error);
+    }
+  }, [garage_id, date, time, dispatch])
+
+  // const fetchReservations = useCallback(async () => {
```

Comment:
If this code isn't necessary I think you should just delete it. You can do this in a future PR if you're testing something first and aren't sure if you're getting rid of it.

Reply:
Good point!